### PR TITLE
Add support for Laravel 5.3

### PR DESCRIPTION
**Pull request by Kaukov**

### CODE REVIEW

**Comment by mmochetti** [comment]
Laravel 5.3 is officially out. Would be great to have this PR merged as we want to craft a 5.3 app with Dingo.

File: composer.json [hunk 1]
```diff
@@ -18,20 +18,20 @@
...
-        "laravel/lumen-framework": "5.1.* || 5.2.*",
+        "illuminate/auth": "5.1.* || 5.2.* || 5.3.*",
+        "illuminate/cache": "5.1.* || 5.2.* || 5.3.*",
+        "illuminate/console": "5.1.* || 5.2.* || 5.3.*",
+        "illuminate/database": "5.1.* || 5.2.* || 5.3.*",
+        "illuminate/events": "5.1.* || 5.2.* || 5.3.*",
+        "illuminate/filesystem": "5.1.* || 5.2.* || 5.3.*",
+        "illuminate/log": "5.1.* || 5.2.* || 5.3.*",
+        "illuminate/pagination": "5.1.* || 5.2.* || 5.3.*",
+        "laravel/lumen-framework": "5.1.* || 5.2.* || 5.3.*",
```
Comment (hskrasek) [inline composer.json]: Lumen hasn't been updated to 5.3 yet.

**Comment by hskrasek** [comment]
Definitely something I'm interested in, but currently want to nail down the issue people are having in regards to 5.1, before adding another version to support.

**Comment by crynobone** [comment]
@hskrasek my company would need to upgrade to Laravel 5.3 within this 1 to 2 week because we need to start supporting `laravel/passport` (which only support 5.3). I would very much like to **NOT** have to maintain a fork to handle this.

**Comment by crynobone** [comment]
I have upgraded to Laravel 5.3

Everything looks fine so far, except this PR need to include https://github.com/laravie/api/commit/51f9c8553efc00b96e552d3c5f281d97272f3e54 due to changes with Laravel Router.

**Comment by hskrasek** [comment]
@Kaukov If you could make the change suggested by @crynobone, I'll give this a test soon and merge if we are all good.

If you're busy though, I can make the change myself and merge in the PR from there.

**Comment by Kaukov** [comment]
@hskrasek is it ok now?

**Comment by jakubkratina** [comment]
Hello guys, what should I do to get to work laravel 5.3 with Dingo? I'm getting this error message (gatherRouteMiddlewares..). Thank you.

**Comment by bkintanar** [comment]
yebah!

**Comment by crynobone** [comment]
@hskrasek still require https://github.com/dingo/blueprint/pull/43 to be merged in order to allow developer to install on L5.3

**Comment by DesmondPang** [comment]
still not work 

I used php composer.phar require dingo/api:v1.0.*@dev

  Problem 1
    - dingo/api v1.0.0-beta5 requires dingo/blueprint 0.2.* -> satisfiable by dingo/blueprint[0.2.x-dev] but these conflict with your requirements or minimum-stability.
    - dingo/api v1.0.0-beta4 requires dingo/blueprint 0.2.* -> satisfiable by dingo/blueprint[0.2.x-dev] but these conflict with your requirements or minimum-stability.
    - dingo/api 1.0.x-dev requires dingo/blueprint 0.2.* -> satisfiable by dingo/blueprint[0.2.x-dev] but these conflict with your requirements or minimum-stability.
    - Conclusion: don't install dingo/api v1.0.0-beta3
    - Installation request for phpdocumentor/reflection-docblock (locked at 3.1.0) -> satisfiable by phpdocumentor/reflection-docblock[3.1.0].
    - Conclusion: remove laravel/framework v5.3.4
    - Conclusion: don't install laravel/framework v5.3.4
    - dingo/api v1.0.0-beta1 requires illuminate/routing 5.1.* -> satisfiable by illuminate/routing[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.28, v5.1.30, v5.1.31, v5.1.41, v5.1.6, v5.1.8].
    - dingo/api v1.0.0-beta2 requires illuminate/routing 5.1.* -> satisfiable by illuminate/routing[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.28, v5.1.30, v5.1.31, v5.1.41, v5.1.6, v5.1.8].
    - don't install illuminate/routing v5.1.1|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.13|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.16|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.2|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.20|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.22|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.25|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.28|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.30|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.31|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.41|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.6|don't install laravel/framework v5.3.4
    - don't install illuminate/routing v5.1.8|don't install laravel/framework v5.3.4
    - Installation request for laravel/framework (locked at v5.3.4, required as 5.3.*) -> satisfiable by laravel/framework[v5.3.4].
    - Installation request for dingo/api v1.0.*@dev -> satisfiable by dingo/api[1.0.x-dev, v1.0.0-beta1, v1.0.0-beta2, v1.0.0-beta3, v1.0.0-beta4, v1.0.0-beta5].

**Comment by hskrasek** [comment]
@DesmondPang That is because I have not tagged a new beta yet. I'm probably going to soon, but if you want to try it out before than, switch to installing dev-master

**Comment by crynobone** [comment]
```
"require": {
    "dingo/api": "dev-master"
    
},
"prefer-stable": true,
"minimum-stability": "dev"
```

**Comment by DesmondPang** [comment]
Thank you for your Beta 6 release. but I still got problem with dingo/blueprint. Need to release blueprint 0.2???

 Problem 1
    - Installation request for dingo/api v1.0.0-beta6 -> satisfiable by dingo/api[v1.0.0-beta6].
    - dingo/api v1.0.0-beta6 requires dingo/blueprint 0.2.* -> satisfiable by dingo/blueprint[0.2.x-dev] but these conflict with your requirements or minimum-stability.

**Comment by hskrasek** [comment]
@DesmondPang Try it once more, looks like the composer file has been pointed to a tag pattern thats never existed.

**Comment by DesmondPang** [comment]
@hskrasek successful. Thank you so much.

**Comment by jakubkratina** [comment]
@hskrasek are you going to tag a new beta soon (in matter of days)? I have big refactoring on the table but I want to create tests first, but can't due to error :) Thank you. Btw. can I help you somehow?

**Comment by bkintanar** [comment]
@jakubkratina `v1.0.0-beta6` has been tagged an hour or so ago.

**Comment by jakubkratina** [comment]
@bkintanar: I can see that now, just updated the composer. Great work, thank you. Can I contribute to dingo development?

**Comment by bkintanar** [comment]
@jakubkratina opensource is always open for contribution. 👍

**Comment by jakubkratina** [comment]
I would like to, is here a contribution document? I have found only slack channel. Thank you.

**Comment by bkintanar** [comment]
here: https://github.com/dingo/api/blob/master/CONTRIBUTING.md

**Comment by pedroskakum** [comment]
Plans for passport? #1159

**Pull request merged**